Port ancillary jobs from Travis to github actions

thomasspriggs opened the pull request:
These were formerly a Travis CI jobs. Setting them up using Github actions gives us a user interface which integrates into the rest of github better. This is a step towards having all jobs setup in Github actions and removing Travis.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

hannes-steffenhagen-diffblue commented on .github/workflows/pull-request-checks.yaml:
```diff
@@ -4,6 +4,44 @@ on:
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
```
To get this to work you’d want `fetch_depth: 0` here to check out the entire repository rather than create a shallow clone (the default).

thomasspriggs committed:
Add github actions clang-format job

This was formerly a Travis CI job. Setting it up using Github actions
gives us a user interface which integrates into the rest of github
better. This is a step towards having all jobs seup in Github actions
and removing Travis.

thomasspriggs committed:
Add github actions linting check

This was formerly a Travis CI job. Setting it up using Github actions
gives us a user interface which integrates into the rest of github
better. This is a step towards having all jobs seup in Github actions
and removing Travis.

thomasspriggs committed:
Add string-table check in github actions

This was formerly a Travis CI job. Setting it up using Github actions
gives us a user interface which integrates into the rest of github
better. This is a step towards having all jobs seup in Github actions
and removing Travis.

hannes-steffenhagen-diffblue commented on .github/workflows/pull-request-check-clang-format.sh:
```diff
@@ -0,0 +1,38 @@
+git diff > formatted.diff
+if [[ -s formatted.diff ]] ; then
+  echo 'Formatting error! The following diff shows the required changes'
+  echo 'Use the raw log to get a version of the diff that preserves spacing'
```
Is this referring to how the github action log presents this?

thomasspriggs replied:
This was just cut and paste from the Travis version. I am not sure if viewing in github actions preserves spacing or not.

hannes-steffenhagen-diffblue commented on .github/workflows/pull-request-check-cpplint.sh:
```diff
@@ -0,0 +1,25 @@
+
+# Find the commit on which the PR is based.
+MERGE_BASE=$(git merge-base ${BASE_BRANCH} ${MERGE_BRANCH})
+echo "Checking standards of code touched since $MERGE_BASE"
```
Something which would be _really_ cool is if we could use https://github.com/marketplace/actions/annotations-action to display inline comments with the needed changes (not an action for this PR, just food for thought for follow-up work).

hannes-steffenhagen-diffblue commented on .github/workflows/pull-request-check-cpplint.sh:
```diff
@@ -0,0 +1,25 @@
+echo "Checking standards of code touched since $MERGE_BASE"
+
+# Do the checking.
+script_folder='./scripts'
```
I feel like this variable isn’t adding a ton of value here.

hannes-steffenhagen-diffblue reviewed (approved):
Looks OK to me

thomasspriggs committed:
Remove redundant use of `script_folder` environment variable
